src/App.js

```diff
@@ -10,25 +10,30 @@ class App extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      authorization: {}
+      authenticationValues: {}
     }
   }
 
   render() {
     return (
       <MuiThemeProvider>
         <div className="App">
-          <LoginWithAmazon setAuthorization={(authResponse) =>
-            this.setState({authorization: {access_token: authResponse.access_token, expires_in: authResponse.expires_in}})} />
+          <LoginWithAmazon setAuthorization={(authResponse) => this.setAuthenticationValues(authResponse)} />
```

Comment:
Why don't you name the parameter authenticationValues too instead of authResponse so that it is in line with the setter name ?

Reply:
authResponse is the response handed over by LoginWithAmazon. authenticationValues state only contains few fields from authResponse. setter receives authResponse and sets authenticationValues state.